tests/legacy-cli/e2e/tests/commands/cache/cache-info.ts

```diff
@@ -0,0 +1,65 @@
+import { execAndWaitForOutputToMatch } from '../../../utils/process';
+import { updateJsonFile } from '../../../utils/project';
+
+export default async function () {
+  const originalCIValue = process.env['CI'];
+
+  try {
+    // Should be enabled by default for local builds.
+    await configureTest('0' /** envCI */);
+    await execAndWaitForOutputToMatch(
+      'ng',
+      ['cache', 'info'],
+      /Effective status on current machine: enabled/,
+    );
+
+    // Should be disabled by default for CI builds.
+    await configureTest('1' /** envCI */, { enabled: true });
+    await execAndWaitForOutputToMatch(
+      'ng',
+      ['cache', 'info'],
+      /Effective status on current machine: disabled/,
+    );
+
+    // Effective status should be enabled when 'environment' is set to 'all' or 'ci'.
+    await configureTest('1' /** envCI */, { environment: 'all' });
+    await execAndWaitForOutputToMatch(
+      'ng',
+      ['cache', 'info'],
+      /Effective status on current machine: enabled/,
+    );
+
+    // Effective status should be enabled when 'environment' is set to 'ci' and run is in ci
+    await configureTest('1' /** envCI */, { environment: 'ci' });
+    await execAndWaitForOutputToMatch(
+      'ng',
+      ['cache', 'info'],
+      /Effective status on current machine: enabled/,
+    );
+
+    // Effective status should be disabled when 'enabled' is set to false
+    await configureTest('1' /** envCI */, { environment: 'all', enabled: false });
+    await execAndWaitForOutputToMatch(
+      'ng',
+      ['cache', 'info'],
+      /Effective status on current machine: disabled/,
+    );
```

Comment:
Should we include tests for `environment: 'local'` with `CI=0` and `CI=1`?

Reply:
Added